habushu-maven-plugin/src/main/java/org/technologybrewery/habushu/util/ContainerizeDepsDockerfileHelper.java

```diff
@@ -0,0 +1,108 @@
+    public static final String ANCHOR_DIRECTORY = "ANCHOR_DIRECTORY";
+    public static final String FINAL_STAGE_TEMPLATE = "dockerfile_final_stage_template";
+    public static final String BUILDER_STAGE_TEMPLATE = "dockerfile_builder_stage_template";
+
+    /**
+     * Create the container stage content with given template, anchor directory and the single module's base directory
+     * @param template container stage template name
+     * @param anchorDirectory the anchor directory
+     * @param moduleBaseDir the module base directory
+     * @return container stage content
+     */
+    public static String createContainerStageContentFrom(String template, String anchorDirectory, String moduleBaseDir) {
+        StringBuilder content = new StringBuilder();
+        InputStream inputStream = ContainerizeDepsDockerfileHelper.class.getClassLoader().getResourceAsStream(template);
+
+        try (InputStreamReader inputStreamReader = new InputStreamReader(inputStream); BufferedReader buffer = new BufferedReader(inputStreamReader)) {
+            String line = buffer.readLine();
+
+            while (line != null) {
+                line = line.strip();
+                if (line.contains(ANCHOR_DIRECTORY)) {
+                    line = line.replaceAll(ANCHOR_DIRECTORY, anchorDirectory);
+                }
+                if (line.contains(REPLACE_WITH_SINGLE_REPO_PROJECT_DIR)) {
+                    line = line.replaceAll(REPLACE_WITH_SINGLE_REPO_PROJECT_DIR, moduleBaseDir);
+                }
+                content.append(line).append("\n");
+                line = buffer.readLine();
+            }
+            inputStream.close();
+        } catch (IOException e) {
+            throw new HabushuException("Could not read from file.", e);
+        }
+        return content.toString();
+    }
+
+    /**
+     * Update the Dockerfile with container stage logic
+     * @param dockerFile the Dockerfile to be updated
+     * @param anchorDirectory the anchor directory
+     * @param moduleBaseDir the module base directory
+     * @return updated Dockerfile content
+     */
+    public static String updateDockerfileWithContainerStageLogic(File dockerFile, String anchorDirectory, String moduleBaseDir) {
+        String builderStageContent = ContainerizeDepsDockerfileHelper.createContainerStageContentFrom(BUILDER_STAGE_TEMPLATE, anchorDirectory, moduleBaseDir);
+        String finalStageContent = ContainerizeDepsDockerfileHelper.createContainerStageContentFrom(FINAL_STAGE_TEMPLATE, null, null);
+        StringBuilder content = new StringBuilder();
+
+        boolean skipLine = false;
+        try (BufferedReader buffer = new BufferedReader(new FileReader(dockerFile))) {
+            String line = buffer.readLine();
+
+            while (line != null) {
+                line = line.strip();
+
+                // start skipping the line if reads HABUSHU_COMMENT_START
+                if (!skipLine && line.contains(HABUSHU_COMMENT_START)) {
```

Comment:
So this means we're not going to modify the code if we encounter `- HABUSHU GENERATED CODE (DO NOT MODIFY)`. I would have thought we still want to overwrite this code in case some changes are made to the underlying monorepo and it's dependencies. 

In short, I thought we always wanted to overwrite the habushu generated code, is that not the case?

Reply:
yes, the previous updated code will be removed and updated again with latest code